### ISSUE TITLE
support jsdoc default value descriptors

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#373]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The link to the original issue is:
https://code.google.com/p/closure-compiler/issues/detail?id=292

**Comment by froodian** [comment]
This lack of support prevents me from using the checkTypes compiler flag

**Comment by dimvar** [comment]
I doubt that we'll support this, especially since ES6 provides default parameters in the language syntax.

**Comment by froodian** [comment]
well at least it's a definitive answer - thanks

**Comment by woldie** [comment]
Would be helpful when using JavaScript codebases that use the default value syntax extensively, like lodash.  I see lodash doing the following:

 * @param {Function} [iteratee=_.identity] The function invoked per iteration.

where ClosureCompiler can successfully compile, but with a warning like this:

```
./node_modules/lodash-compat/utility/times.js:49: WARNING - actual parameter 1 of bindCallback$$module$home$node_modules$lodash_compat$utility$times does not match formal parameter
found   : (Function|null|undefined)
required: (Function|null)
  iteratee = bindCallback(iteratee, thisArg, 1);
```

Is it possible to suppress this warning on certain 3rd party folders and not my 1st party code (since I will obey CC's JSDoc conventions).  Lodash is throwing hundreds of these warnings at me.  For now, I will try grepping CC's output and filter out any warnings from the lodash compilation.

I'm not keen on disabling verbose warnings and checkTypes altogether as @froodian is doing since those warnings are one of the main benefits of CC for me.
